### ISSUE TITLE
Unlabeled image pesudo label problem

### ISSUE THREAD

**Issue by SeanCho1996** [marwankefah/mean-teacher-medical-imaging#1]
Hello Marawan, this repo is a great implementation of Mean-teacher in medical image segmentation, it's really hard to find one implementation within the new version of pytorch in segmentation domain.

Yet I have just one question about how you deal with the unlabeled samples. I noticed that you create a zero mask for these samples, but in loss computation part, you directly compute the CE loss of model output and this zero mask, isn't this a problem here? The model will learn to predict a zero mask, no?

I am not sure if I have misunderstood your code, or you have an other processing of unlabeled samples that I didn't notice.

Thanks in advance

**Comment by marwankefah** [comment]
Hi Sean, thanks a lot for reaching for me,  args.labeled_bs that is specified in the config file is responsible for the number of labeled batches that will be fed to the network every iteration. while calculating the CEloss ce_loss(outputs[:args.labeled_bs],
                              label_batch[:][:args.labeled_bs].long()) so I just use the labeled batches returned by the TwoBatchSampler.

let me know if there is smth wrong, and i will be happy to fix

**Comment by SeanCho1996** [comment]
Ah I see your point, so this TwoBatchSampler can be used to balance the self-training and consistency at the same time. It 's a great design, thanks a lot!

**Comment by marwankefah** [comment]
Yes, and it is similar to https://github.com/HiLab-git/SSL4MIS/ repository but with torchio 
Thank you,
